File: 70. Climbing Stairs/py/climbStairs.py
# LeetCode problem 70. Climbing Stairs
# <https://leetcode.com/problems/climbing-stairs/description/>

class Solution:
    def climbStairs(self, n: int) -> int:
        return self.fibonacci(n)

    def fibonacci(self, n: int) -> int:
        if n <= 0:
            raise ValueError()
        elif n == 1:
            return 1
        else:
            a = 1
            b = 2
        for _ in range(3, n + 1):
            c = a + b
            a = b
            b = c
        return b

File: 70. Climbing Stairs/cs/ClimbStairs/ClimbStairs.cs
// LeetCode problem 70. Climbing Stairs
// https://leetcode.com/problems/climbing-stairs/description/

namespace LeetCode.ClimbStairs;

public class Solution
{
    public int ClimbStairs(int n)
    {
        return Fibonacci(n);
    }

    private static int Fibonacci(int n)
    {
        // LeetCode guarantees that n >= 1
        if (n == 1)
            return 1;
        if (n == 2)
            return 2;

        int a = 1, b = 2;
        for (int i = 3; i <= n; i++)
            (a, b) = (b, a + b);

        return b;
    }
}